### PR TITLE
Roll up org stats, filter out internal metrics

### PR DESCRIPTION
**Pull request by kpthatsme**
## Changes
- Aggregates the team-level stats making it easier to see how the org as a whole is doing
- Added the `organization__for_internal_metrics=True` exclusion to the teams query 
## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: posthog/tasks/status_report.py [hunk 1]
```diff
@@ -49,10 +49,20 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
         (plugin_config.plugin.name for plugin_config in plugin_configs if plugin_config.enabled)
     )
 
-    for team in Team.objects.all():
+    org_usage_summary: Dict[str, Any] = {
```
Comment (macobo) [inline posthog/tasks/status_report.py:52]: Nit: It's `Dict[str, int]`

It's imo important not to mess up terminology. We have 3 layers of abstraction:

instance > organization > team.

An instance may have multiple orgs, so this should be named `instance_usage_summary`

**Review by macobo — CHANGES_REQUESTED** [review]

**Comment by macobo** [comment]
lgtm (thanks) other than the 1 comment

File: posthog/tasks/status_report.py [hunk 2]
```diff
@@ -74,19 +89,21 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
...
             team_report["dashboards_count"] = team_dashboards.count()
+            org_usage_summary["dashboards_created"] += team_report["dashboards_count"]
             team_report["dashboards_template_count"] = team_dashboards.filter(creation_mode="template").count()
             team_report["dashboards_shared_count"] = team_dashboards.filter(is_shared=True).count()
             team_report["dashboards_tagged_count"] = team_dashboards.exclude(tags=[]).count()
 
             # Feature Flags
             feature_flags = FeatureFlag.objects.filter(team=team).exclude(deleted=True)
             team_report["ff_count"] = feature_flags.count()
+            org_usage_summary["feature_flags_created"] += team_report["ff_count"]
```
Comment (macobo) [inline posthog/tasks/status_report.py:100]: Perhaps worth keeping the name same even if it's bad? :) Just for sake of easy analysis

File: posthog/tasks/status_report.py [hunk 3]
```diff
@@ -74,19 +89,21 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
...
             team_report["ff_active_count"] = feature_flags.filter(active=True).count()
-
             report["teams"][team.id] = team_report
         except Exception as err:
             capture_event("instance status report failure", {"error": str(err)}, dry_run=dry_run)
 
+    report["usage_summary"] = org_usage_summary
```
Comment (paolodamico) [inline posthog/tasks/status_report.py:106]: Naming as @macobo mentioned above

File: posthog/tasks/test/test_status_report.py [hunk 4]
```diff
@@ -45,6 +45,12 @@ def test_team_status_report_event_counts(self) -> None:
             self.assertEqual(team_report["persons_count_new_in_period"], 2)
             self.assertEqual(team_report["persons_count_active_in_period"], 1)
 
+            usage_summary = status_report(dry_run=True).get("usage_summary")
+            self.assertEqual(usage_summary["events_used_in_period"], 3)  # type: ignore
```
Comment (paolodamico) [inline posthog/tasks/test/test_status_report.py:49]: I think it's worth adding events to a separate org and making sure these are properly accounted. In addition I would also add events on an _internal metrics_ org to make sure those are not counted. Think it's worth adding more strenuous tests as we'll use this for billing.

Not specifically related with the goal of this PR but I would also add an assertion/test for edge cases of events in the boundaries of periods, to make sure they're not double counted in two reports.

File: posthog/tasks/status_report.py (hunk 1 above)
Comment (kpthatsme) [reply posthog/tasks/status_report.py]: Thank you for clarifying the layers!

File: posthog/tasks/test/test_status_report.py (hunk 4 above)
Comment (kpthatsme) [reply posthog/tasks/test/test_status_report.py]: Agreed, added some cases for events outside of the period as well as some on an internal metrics team

**Comment by kpthatsme** [comment]
Thanks @paolodamico @macobo for the review – updated some names and added a few more tests

**Review by macobo — APPROVED** [review]
Looking great!

**Pull request merged**